algolia/search/dictionary_entry_compound.go

```diff
@@ -0,0 +1,51 @@
+package search
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Compound represents an entry in the Compounds dictionary.
+type Compound struct {
+	objectID      string
+	language      string
+	Word          string
+	Decomposition []string
+}
+
+func (c Compound) ObjectID() string { return c.objectID }
+func (c Compound) Language() string { return c.language }
+
+// NewCompound is a constructor of Compound
+func NewCompound(objectID, language string, word string, decomposition []string) Compound {
+	return Compound{objectID, language, word, decomposition}
+}
+
+type compound struct {
+	ObjectID      string   `json:"objectID"`
+	Language      string   `json:"language"`
+	Word          string   `json:"word"`
+	Decomposition []string `json:"decomposition"`
+}
+
+func (c Compound) MarshalJSON() ([]byte, error) {
+	return json.Marshal(compound{c.objectID, c.language, c.Word, c.Decomposition})
+}
+
+func (c *Compound) UnmarshalJSON(data []byte) error {
+	if string(data) == jsonNull {
```

Comment:
If the unmarshaled json byte is actually `null`, shouldn't we have a nil *Compound instead of keeping the current one?

If not, we should at least "nil" the current instance properties as they would remain untouched. I'm aware that it's unlikely someone will unmarshal in an already initialized instance, but I think it highlights the probable need to get a `nil` pointer for unmarshaling a `null`.

Reply:
I'm not sure if I understand what do you mean by: 
> have a nil *Compound instead of keeping the current one
Not just return `nil`, but set all the properties to their zero-values/nil? 

Same thing, I borrowed this logic from the [Synonyms unmarshalling code](https://github.com/algolia/algoliasearch-client-go/blob/c20c1ad93f065ffa3e27db42003baed6f08fc437/algolia/search/synonym_alt_correction_1.go#L32) considering it as the best practice.

Reply:
👌 keeping it in line with the rest of the implem makes sense.